synapse/util/caches/lrucache.py

```diff
@@ -568,6 +591,42 @@ def cache_get(
                     metrics.inc_misses()
                 return default
 
+        @overload
+        def cache_get_multi(
+            key: tuple,
+            default: Literal[None] = None,
+            update_metrics: bool = True,
+        ) -> Union[None, TreeCacheNode]:
+            ...
+
+        @overload
+        def cache_get_multi(
+            key: tuple,
+            default: T,
+            update_metrics: bool = True,
+        ) -> Union[T, TreeCacheNode]:
+            ...
+
+        @synchronized
+        def cache_get_multi(
+            key: tuple,
+            default: Optional[T] = None,
+            update_metrics: bool = True,
+        ) -> Union[None, T, TreeCacheNode]:
+            """Used only for `TreeCache` to fetch a subtree."""
```

Comment:
... but what does it actually do? What are the arguments, what are the return values?

also it's not used by TreeCache; it is used to expose a `get_multi` method when the backing cache is a `TreeCache`.

Reply:
... also, how is it different to a simple `cache_get` with `update_last_access=False`?